### ISSUE TITLE
Data Filterers are not working correctly with uppercase letters

### ISSUE THREAD

**Issue by ViliusBa** [iTwin/appui#545]
**Describe the bug**

DataFilterer does not match entries when uppercase letters are used. For instance, filtering with lowercase letters provides such results:
![image](https://github.com/iTwin/appui/assets/114080994/6ea9d57b-21b7-4480-b512-5da0db4a192f)

while filtering with uppercase does not match anything, even though it should:
![image](https://github.com/iTwin/appui/assets/114080994/48f8c7a9-6864-4ca5-97c1-e13fd4960f20)

This does not seem to be intended behavior, because the filter is case insensitive when lowercase letters are used, whereas using uppercase ones does not provide any matches.

**To Reproduce**

Filter setup: 
```
function getFilter(filterText: string){
  const valueFilterer = new DisplayValuePropertyDataFilterer(filterText);
  const labelFilterer = new LabelPropertyDataFilterer(filterText);
  const categoryFilterer = new PropertyCategoryLabelFilterer(filterText);
  const valueAndRecordFilterer = new CompositePropertyDataFilterer(valueFilterer, CompositeFilterType.Or, labelFilterer);

  return new CompositePropertyDataFilterer(valueAndRecordFilterer, CompositeFilterType.Or, categoryFilterer);
}
```

**Expected behavior**

Filtering shouldn't be case sensitive, or there should be an option to determine whether it is or not.

**AppUI versions**

```
 "@itwin/appui-abstract": "^4.0.0",
 "@itwin/appui-react": "^4.3.0",
```

**Comment by maheshchandra10** [comment]
Please let me know if I can work on this.

**Comment by ViliusBa** [comment]
Sure, go ahead

**Comment by grigasp** [comment]
> Hi @ViliusBa, @grigasp and @jjbeckman13. The fix for this was released today as 4.7.1, let us know if this fixes your issue!

I can confirm this is fixed in [vcr](https://github.com/iTwin/viewer-components-react) test app.
